tests/test_datasets/test_dataset_functions.py

```diff
@@ -1370,12 +1375,28 @@ def test_data_edit(self):
 
         # Case 2
         # only owners (or admin) can edit all critical fields of datasets
-        # this is a dataset created by CI, so it is editable by this test
-        did = 315
-        result = edit_dataset(did, default_target_attribute="col_1", ignore_attribute="col_2")
+        # for this, we need to first clone a dataset to do changes
+        did = fork_dataset(1)
```

Comment:
It would probably also make sense to test each case in a separate unit test? I don't really see a reason to test both scenarios in the same test, and it would help both with separating concerns and load balancing.

Reply:
Good catch, will change this.